packages/astro/src/core/build/plugins/plugin-hoisted-scripts.ts

```diff
@@ -73,23 +72,18 @@ export function vitePluginHoistedScripts(
 					shouldInlineAsset(output.code, output.fileName, assetsInlineLimit);
 				let removeFromBundle = false;
 				const facadeId = output.facadeModuleId!;
-				const pages = internals.hoistedScriptIdToPagesMap.get(facadeId)!;
-				for (const pathname of pages) {
-					const vid = viteID(new URL('.' + pathname, settings.config.root));
-					const pageInfo = getPageDataByViteID(internals, vid);
-					if (pageInfo) {
-						if (canBeInlined) {
-							pageInfo.hoistedScript = {
-								type: 'inline',
-								value: output.code,
-							};
-							removeFromBundle = true;
-						} else {
-							pageInfo.hoistedScript = {
-								type: 'external',
-								value: id,
-							};
-						}
+				for (const pageData of getPageDatasByHoistedScriptId(internals, facadeId)) {
```

Comment:
As you noticed from the Github bot comments, `settings` isn't used anymore, so we can remove it

Reply:
Removed. Thanks.